chain/client/src/view_client.rs

```diff
@@ -527,26 +527,31 @@ impl Handler<GetChunk> for ViewClientActor {
                 ))
             })
         };
-        match msg {
-            GetChunk::ChunkHash(chunk_hash) => self.chain.get_chunk(&chunk_hash).map(Clone::clone),
+
+        let chunk = match msg {
+            GetChunk::ChunkHash(chunk_hash) => {
+                self.chain.get_chunk(&chunk_hash).map(Clone::clone)?
+            }
             GetChunk::BlockHash(block_hash, shard_id) => {
                 let block = self.chain.get_block(&block_hash).map(Clone::clone);
-                get_chunk_from_block(block, shard_id, &mut self.chain)
+                get_chunk_from_block(block, shard_id, &mut self.chain)?
```

Comment:
I suggest we pass the block itself instead of a result of a block into the `get_chunk_from_block` helper function (please, update the function to reflect this change), so the usage would look like:

```rust
let block = self.chain.get_block(&block_hash)?.clone();
get_chunk_from_block(block, shard_id, &mut self.chain)?
```